app/src/main/java/de/danoeh/antennapod/activity/MediaplayerActivity.java

```diff
@@ -457,7 +475,30 @@ protected void setupGUI() {
         setContentView(getContentViewResourceId());
         sbPosition = (SeekBar) findViewById(R.id.sbPosition);
         txtvPosition = (TextView) findViewById(R.id.txtvPosition);
+
+        SharedPreferences prefs = getSharedPreferences(PREFS, MODE_PRIVATE);
+        showTimeLeft = prefs.getBoolean(PREF_SHOW_TIME_LEFT,false);
+        Log.d("timeleft",showTimeLeft? "true":"false");
         txtvLength = (TextView) findViewById(R.id.txtvLength);
+        txtvLength.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showTimeLeft = !showTimeLeft;
+                Playable media = controller.getMedia();
```

Comment:
I believe it's possible for media to be null. Need to check for that or we'll get an NPE.